include/tvm/runtime/object.h

```diff
@@ -890,7 +890,7 @@ inline bool Object::IsInstance() const {
       // The condition will be optimized by constant-folding.
       if (TargetType::_type_child_slots != 0) {
         uint32_t end = begin + TargetType::_type_child_slots;
-        if (self->type_index_ >= begin && self->type_index_ < end) return true;
+        if (self->type_index_ >= begin && self->type_index_ <= end) return true;
```

Comment:
thanks, for readability, please change to uint32_t end = begin + TargetType::_type_child_slots +1;